### PR TITLE
feat: Send fiat amount

### PR DESCRIPTION
**Pull request by maxwellmattryan**
# Description of change

- Adds user's chosen fiat currency to list of denominations in dropdown menu for when sending a transaction
- Adds more functions to core lib with tests: `convertFromFiat` and `isFiatCurrency`
- Adds a constant, `DUST_THRESHOLD`, indicating what is considered dust

## Links to any relevant issues

- #763
- #1168 

## Type of change

- New (a change which implements a new feature)

## How the change has been tested

__Both software and Ledger profiles__ have been tested with this change.

Tested on:

- MacOS (Catalina 10.15.7)

## Change checklist

- [x] I have followed the contribution guidelines for this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by nicole-obrien** [comment]
@maxwellmattryan Just reviewing your PR:
When sending IOTA, if I enter the an amount in IOTA that is more than the supply then amount is truncated to the maximum supply. If I am entering a fiat amount and I send an amount that is really big then the application semi-freezes and when I delete the amount the subtext stays there and I am unable to navigate to anything else within the application.

**Review by nicole-obrien — CHANGES_REQUESTED** [review]
Added individual comments, but let's test this on ledger before approval.

File: packages/shared/routes/dashboard/wallet/views/Send.svelte [hunk 1]
```diff
@@ -269,20 +269,38 @@
...
+        /**
+         * NOTE: Sometimes max values from fiat calculations 
+         * aren't precise enough, so we have to ensure that the
+         * actual max amount is applied when the user clicks
+         * the button.
+        */
+
+        let isFiat = isFiatCurrency(unit)
+        let isMaxAmount = amount === convertToFiat(from.balance, $currencies[CurrencyTypes.USD], $exchangeRates[unit]).toString()
+        let hasDustRemaining = Math.abs(from.balance - _amount) < 1_000_000
```
[the remark this answers is not in the document]
Comment (maxwellmattryan) [reply packages/shared/routes/dashboard/wallet/views/Send.svelte:299]: Have defined a constant `DUST_THRESHOLD` for in `currency.ts` ~

File: packages/shared/components/inputs/Amount.svelte [hunk 2]
```diff
@@ -3,55 +3,133 @@
...
+            return formatUnitBestMatch(rawAmount)
+        }
+
+        return convertAmount(_amount, unit, _convert)
+    }
+
+    const convertAmount = (_amount, _unit, convertFn) => {
+        if(!amount) return null
+
+        console.log('AMT: ', _amount)
```
[the remark this answers is not in the document]
Comment (maxwellmattryan) [reply packages/shared/components/inputs/Amount.svelte]: Yeah 😅

File: packages/shared/components/inputs/Amount.svelte [hunk 3]
```diff
@@ -3,55 +3,133 @@
...
         formatCurrency,
+        isFiatCurrency,
         parseCurrency,
         replaceCurrencyDecimal,
     } from 'shared/lib/currency'
     import { activeProfile } from 'shared/lib/profile'
-    import { changeUnits, formatUnitPrecision, UNIT_MAP } from 'shared/lib/units'
+    import { changeUnits, formatUnitBestMatch, formatUnitPrecision, UNIT_MAP } from 'shared/lib/units'
+
+    type AmountUnit = Unit | string
```
[the remark this answers is not in the document]
Comment (maxwellmattryan) [reply packages/shared/components/inputs/Amount.svelte]: Change has been made!

**Comment by maxwellmattryan** [comment]
Thanks for reviewing, @nicole-obrien. I've made some changes and the PR is good to be merged given your approval!

**Review by nicole-obrien — COMMENTED** [review]
Did you test the changes on desktop and ledger?

**Comment by maxwellmattryan** [comment]
@nicole-obrien 

> Did you test the changes on desktop and ledger?

Yep, they're good to go!

**Review by nicole-obrien — APPROVED** [review]

**Pull request merged**